server/src/main/java/keywhiz/KeywhizService.java

```diff
@@ -76,7 +76,13 @@
    * @throws Exception for any uncaught application exception
    */
   public static void main(String[] args) throws Exception {
-    new KeywhizService().run(args);
+    logger.info("Starting Keywhiz...");
+    try {
+      new KeywhizService().run(args);
+    } catch (Exception e) {
+      logger.error("Exiting due to uncaught exception", e);
+      throw e;
+    }
   }
```

Comment:
We had an issue where database validation failure was causing Keywhiz to enter a death spiral. Unforutnately, without a top-level catch clause and logging statement, the stack trace was ending up in the nanny process's logs rather than the actual application logs. This change aims to ensure that any exception which causes Keywhiz to exit will end up in the application logs. It also adds an explicit starting statement to make obvious where each loop in the spiral begins and ends.